File: 202 Happy Number/Java/Solution.java
import java.util.ArrayList;
import java.util.List;

public class Solution {
    private static int getProcessedNum(int n) {
        int num = 0;
        while (n != 0) {
            num += Math.pow(n % 10, 2);
            n /= 10;
        }
        return num;
    }
    
    public boolean isHappy(int n) {
        List<Integer> previous = new ArrayList<>();
        previous.add(n);

        while (true) {
            int num = getProcessedNum(n);
            
            if (num == 1) {
                return true;
            }
            if (previous.contains(num)) {
                return false;
            }

            n = num;
            previous.add(n);
        }
    }
}

File: 202 Happy Number/Python/Solution.py
from typing import List, Self


class Solution:
    def getProcessedNum(self: Self, n: int) -> int:
        num: int = 0
        while (n != 0):
            num += pow(n % 10, 2)
            n //= 10
        return num

    def isHappy(self: Self, n: int) -> bool:
        previous: List[int] = [n]
        while (True):
            num: int = self.getProcessedNum(n)
            if num == 1:
                return True
            if num in previous:
                return False
            n = num
            previous.append(n)